## 1. Create Python class named `Student` and display its type

In [104]:
# CODE HERE
class Student:
    pass
print(type(Student))

<class 'function'>


<class 'type'>


## 2. How to call a variable and function by creating a simple class


In [12]:
class Dog:
     #CODE HERE
        def a(self):
            return "This is Variable1"
        def b(self):
            return "I'm from function "
add = Dog()
print(add.a())
print(add.b()+" "+add.a())

This is Variable1
I'm from function  This is Variable1


In [ ]:
class Dog:
     #CODE HERE

This is Variable1
I'm from function This is Variable1


## 3. Create a class with one `static variable` and two `instace variables`

In [27]:
class Demo():
    #CODE HERE
    static = "I am Static variable "
    def __init__(self,var1,var2):
        self.var1 = var1
        self.var2 = var2
        
a= Demo("instance_var1","instance_var2")
print(a.static)
print(a.var1)
print(a.var2)

I am Static variable 
instance_var1
instance_var2


In [11]:
class Demo():
#CODE HERE

I am Static variable
instance_var1
instance_var2


## 4. Create a `Vehicle` class with max_speed and mileage instance attributes

In [77]:
# CODE HERE
class Vehicle:
    def __init__(self,max_speed,mileage):
        self.max_speed= max_speed
        self.mileage= mileage
        
a= Vehicle(240,18)
print(a.max_speed,end=" ")
print(a.mileage)

240 18


In [ ]:
class Vehicle:
    #CODE HERE

240 18


## 5. Create a simple class `Person` with constructer and one argument `name`

In [34]:
# CODE HERE
class Person():
   
    #CODE HERE
    def __init__(self,name):
        self.name = name
        
    def say_hi(self):
        return "Hello, my name is "
    
p = Person('Ram')
print(p.say_hi()+p.name)

Hello, my name is Ram


In [ ]:
# A Sample class with init method
class Person:
   
    #CODE HERE
   
p = Person('Ram')
p.say_hi()

Hello, my name is Ram


## 6. Create a class for `Students` information like `name, rollno, marks` and `school name` with static variable.
- create a function for displaying student information
- create one more function for getting students marks and percentage 

In [85]:
class Student:
    # CODE HERE
    name = "Name: abc"
    rollNo= "RollNo: 1"
    marks= "Marks: 399"
    college= "College: DIT"
    
    def __init__(self,name,rollno,marks,college):
        self.name= name
        self.rollno= rollno
        self.marks= marks
        self.college=college
        
    def students_marks(self):
        print(f"Dictionary :  {dict(name = self.name, rno = self.rollno, marks = self.marks)}")
        
    def percentage(self):
        print(f"percentile of {self.marks} out of 500 is :{self.marks /500*100}") 
        
a= Student("abc",1,399,"DIT")
print(a.name, end=", ")
print(a.rollNo, end=", ")
print(a.marks, end=", ")
print(a.college)
a.students_marks()
a.percentage()

abc, RollNo: 1, 399, DIT
Dictionary :  {'name': 'abc', 'rno': 1, 'marks': 399}
percentile of 399 out of 500 is :79.80000000000001



Name: abc, RollNo: 1, Marks: 399, College: DIT
Dictionary :  {'name': 'abc', 'rno': 1, 'marks': 399}
percentahe of 399 out of 500 is :  79.80000000000001

## 7. Create a child class `Bus` that will inherit all of the variables and methods of the `Vehicle` class

In [94]:
# CODE HERE
class Vehicle:
    
    def __init__(self,vehicle_name,speed,mileage):
        self.vehicle_name = vehicle_name
        self.speed = speed
        self.mileage=mileage
        
class Bus(Vehicle):
    
    def details(self):
        print(f"Vehicle Name: {self.vehicle_name} Bus speed: {self.speed} Mileage: {self.mileage}")
a= Bus("School Bus",180,12)
a.details()

Vehicle Name: School Bus Bus speed: 180 Mileage: 12


In [ ]:
class Vehicle:

    #CODE HERE

Vehicle Name: School Bus Speed: 180 Mileage: 12


## 8. Create a class `Employee` with two arguments `name and salary`
- create a function which displays the employee information
- Print total number of employees 

In [95]:
class Employee:
    def __init__(self,name,salary):
        self.name = name
        self.salary = salary
        
    def Display(self):
        print("Name : ",self.name,", Salary : ",self.salary)
    
    def Display1(self):
        print("Name : ",self.name,", Salary : ",self.salary)
        
E = Employee("Zara",2000)
E1 = Employee("Manni",5000)
Employees = 2
E.Display()
E1.Display1()
print("Total Employee",Employees)

Name :  Zara , Salary :  2000
Name :  Manni , Salary :  5000
Total Employee 2


In [10]:
class Employee:
   
   #CODE HERE

Name :  Zara , Salary:  2000
Name :  Manni , Salary:  5000
Total Employee 2
